### PR TITLE
Ioctl should allow import/export from to keystore file #1191

### PR DESCRIPTION
**Pull request by lzxm160**
work on #1191 

1、
ioctl account import keystore xx UTC--2019-05-23T09-36-39.493806681Z--3ac6ad0777e76ae6974ff89bf3a719adac600077
#xx: Enter your password of keystore, which will not be exposed on the screen.
#xx: Set password
#xx: Enter password again
New account #xx is created. Keep your password, or your will lose your private key.

2、
ioctl account import keystore xx UTC--2019-05-23T09-36-39.493806681Z--3ac6ad0777e76ae6974ff89bf3a719adac600077
#xx: Enter your password of keystore, which will not be exposed on the screen.
Error: could not decrypt key with given passphrase

3、
ioctl account import keystore xxx ./nofile
#xxx: Enter your password of keystore, which will not be exposed on the screen.
Error: keystore file "./nofile" read error

4、
ioctl account import key aa
#aa: Enter your private key, which will not be exposed on the screen.
#aa: Set password
#aa: Enter password again
New account #aa is created. Keep your password, or your will lose your private key.

### CODE REVIEW

File: cli/ioctl/cmd/account/accountimport.go [hunk 1]
```diff
@@ -72,3 +95,38 @@ func accountImport(args []string) (string, error) {
 		"New account #%s is created. Keep your password, or your will lose your private key.",
 		alias), nil
 }
+func accountImportKeyStore(args []string) (string, error) {
```
Comment (zjshen14) [inline cli/ioctl/cmd/account/accountimport.go]: It's almost identical to accountImportKey. Can we try to consolidate the code?

File: cli/ioctl/cmd/account/account.go [hunk 2]
```diff
@@ -196,3 +204,101 @@ func newAccountByKey(alias string, privateKey string, walletDir string) (string,
 	}
 	return addr.String(), nil
 }
+
+func newAccountByKeyStore(alias, passwordOfKeyStore, keyStorePath string, walletDir string) (string, error) {
+	keyJSON, err := ioutil.ReadFile(keyStorePath)
+	if err != nil {
+		return "", fmt.Errorf("keystore file \"%s\" read error", keyStorePath)
+	}
+	key, err := keystore.DecryptKey(keyJSON, passwordOfKeyStore)
```
Comment (zjshen14) [inline cli/ioctl/cmd/account/account.go]: As you already decrypted private key, can you just reuse newAccountByKey() for the following step?

File: cli/ioctl/cmd/account/account.go [hunk 3]
```diff
@@ -196,3 +204,101 @@ func newAccountByKey(alias string, privateKey string, walletDir string) (string,
...
+	if key != nil && key.PrivateKey != nil {
+		defer func(k *ecdsa.PrivateKey) {
```
[the remark this answers is not in the document]
Comment (lzxm160) [reply cli/ioctl/cmd/account/account.go]: clear privatekey in memory

File: cli/ioctl/cmd/account/accountimport.go [hunk 4]
```diff
@@ -22,53 +22,113 @@ import (
...
+	if err := ioutil.WriteFile(config.DefaultConfigFile, out, 0600); err != nil {
+		return "", fmt.Errorf("failed to write to config file %s", config.DefaultConfigFile)
 	}
-	fmt.Printf("#%s: Enter your private key, which will not be exposed on the screen.\n", alias)
-	privateKeyBytes, err := terminal.ReadPassword(int(syscall.Stdin))
+	return fmt.Sprintf(
+		"New account #%s is created. Keep your password, or your will lose your private key.",
+		alias), nil
+}
+func readStdin() (string, error) {
```
Comment (dustinxie) [inline cli/ioctl/cmd/account/accountimport.go]: readPassword() or readPasswordFromStdin()

File: cli/ioctl/cmd/account/account.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (zjshen14) [reply cli/ioctl/cmd/account/account.go]: Can you add the code comment about this?

**Pull request merged**